src/Tribe/Dependency.php

```diff
@@ -525,18 +521,17 @@ public function check_plugin( $main_class ) {
 		 * @return bool  returns false if any dependency is invalid
 		 */
 		protected function check_addon_dependencies( $main_class ) {
-
-			$addon_dependencies = 0;
+			$addon_dependencies = [];
 
 			foreach ( $this->registered_plugins as $registered ) {
 				if ( empty( $registered['dependencies']['addon-dependencies'][ $main_class ] ) ) {
 					continue;
 				}
 
-				$addon_dependencies = $this->has_valid_dependencies( $registered, $registered['dependencies']['addon-dependencies'], true );
+				$addon_dependencies[ $main_class ] = $this->has_valid_dependencies( $registered, $registered['dependencies']['addon-dependencies'], true );
 			}
 
-			return $addon_dependencies;
+			return in_array( true,  $addon_dependencies, true );
```

Comment:
This code essentially means "if at least one of the dependencies is fulfilled, then return true"; why not just `return` from the `foreach` loop?